rest/server_context.go

```diff
@@ -1147,6 +1148,70 @@ func (sc *ServerContext) updateCalculatedStats() {
 
 }
 
+func initClusterAgent(clusterAddress, clusterUser, clusterPass, certPath, keyPath, caCertPath string) (*gocbcore.Agent, error) {
+	authenticator, err := base.GoCBCoreAuthConfig(clusterUser, clusterPass, certPath, keyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	tlsRootCAProvider, err := base.GoCBCoreTLSRootCAProvider(caCertPath)
+	if err != nil {
+		return nil, err
+	}
+
+	config := gocbcore.AgentConfig{
+		Auth:              authenticator,
+		TLSRootCAProvider: tlsRootCAProvider,
+	}
+
+	err = config.FromConnStr(clusterAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	agent, err := gocbcore.CreateAgent(&config)
```

Comment:
If we have an agent, should we be using agent.DoHTTPRequest instead of working with the management endpoints directly?

Reply:
We had this discussion in a Sync up about this last week (as I originally played around with DoHTTPRequest in my PoC) and I was fairly sure we came to the conclusion that we wanted to do that on our side so we can explicitly control the operation.